src/courses/forms.py

```diff
@@ -22,6 +22,9 @@ def __init__(self, *args, **kwargs):
         if courses_qs.count() == 1:
             self.fields['course'].empty_label = None
 
+        block_qs = Block.objects.filter(active=True)
+        self.fields['block'].queryset = block_qs
+
```

Comment:
I think this should be moved above the code that sets a default if there is only one option.  As is, it sets the default based on all Blocks, but then you change the queryset afterwards.  So move this up, and then change Block.objects.count() to block_qs.count()

Reply:
done